examples/IF_curve_LIF.html

```diff
@@ -0,0 +1,311 @@
+                            }
+                        },
+                        titleFont: {
+                            size: 16
+                        },
+                        bodyFont: {
+                            size: 14
+                        }
+                    },
+                    title: {
+                        display: true,
+                        text: 'Neuron Firing Rate vs. Input Current',
+                        font: {
+                            size: 20, 
+                            weight: 'bold'
+                        },
+                        color: '#2b6cb0', 
+                        padding: {
+                            bottom: 15
+                        }
+                    }
+                },
+                scales: {
+                    x: {
+                        title: {
+                            display: true,
+                            text: 'Input Potential (mV)',
+                            font: {
+                                size: 16, 
+                                weight: 'medium'
+                            },
+                            color: '#2c5282' 
+                        },
+                        ticks: {
+                            font: {
+                                size: 14 
+                            },
+                            color: '#4a5568' 
+                        },
+                        grid: {
+                            color: '#ebf4ff' 
+                        }
+                    },
+                    y: {
+                        title: {
+                            display: true,
+                            text: 'Firing Rate (Hz)',
+                            font: {
+                                size: 16, 
+                                weight: 'medium'
+                            },
+                            color: '#2c5282' 
+                        },
+                        ticks: {
+                            font: {
+                                size: 14 
+                            },
+                            color: '#4a5568' 
+                        },
+                        grid: {
+                            color: '#ebf4ff'
+                        },
+                        beginAtZero: true
+                    }
+                }
+            }
+        });
+    }
+    
+    function completeProgress() {
+        if (chartComplete && simComplete) {
+            document.getElementById('brian_progress_bar').style.width = '100%';
+            document.getElementById('progress_animation').style.display = 'none';
+            document.getElementById('status_text').textContent = 'Completed';
+            document.getElementById('run_button').disabled = false;
+        }
+    }
+    
+    var brian_sim = new BrianSimulation(result_plots=[{type: 'custom', func: myplot}]);
+    
+    brian_sim.on_progress = function(progress) {
```

Comment:
I am a bit confused: the `BrianSimulation` object does not seem to have an `on_progress` function, and I don't think this code gets called ?

Maybe have a look into `brian.js` to see how the progress is updated? Please note that the current code might also be suboptimal or even broken – it did not see much testing beyond the existing examples.